src/components/viewmodels/roomlist/RoomListHeaderViewModel.tsx

```diff
@@ -101,20 +100,25 @@ export function useRoomListHeaderViewModel(): RoomListHeaderViewState {
         PosthogTrackers.trackInteraction("WebRoomListHeaderPlusMenuCreateChatItem", e);
     }, []);
 
-    const createRoom = useCallback((e: Event) => {
-        defaultDispatcher.fire(Action.CreateRoom);
-        PosthogTrackers.trackInteraction("WebRoomListHeaderPlusMenuCreateRoomItem", e);
-    }, []);
+    const createRoom = useCallback(
+        (e: Event) => {
+            if (activeSpace) showCreateNewRoom(activeSpace);
+            else defaultDispatcher.fire(Action.CreateRoom);
+            PosthogTrackers.trackInteraction("WebRoomListHeaderPlusMenuCreateRoomItem", e);
+        },
+        [activeSpace],
+    );
 
     const elementCallVideoRoomsEnabled = useFeatureEnabled("feature_element_call_video_rooms");
-    const createVideoRoom = useCallback(
-        () =>
+    const createVideoRoom = useCallback(() => {
+        const type = elementCallVideoRoomsEnabled ? RoomType.UnstableCall : RoomType.ElementVideo;
+        if (activeSpace) showCreateNewRoom(activeSpace, type);
+        else
             defaultDispatcher.dispatch({
                 action: Action.CreateRoom,
                 type: elementCallVideoRoomsEnabled ? RoomType.UnstableCall : RoomType.ElementVideo,
```

Comment:
Why doesn't this just reuse type from L114?

Reply:
Because I'm dumb